### DISCUSSION TITLE
Grove - UV Sensor | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#1283] [Q&A]
# Grove - UV Sensor | Seeed Studio Wiki

Grove - UV Sensor

https://wiki.seeedstudio.com/Grove-UV_Sensor/

**Comment by jmschp** [comment]
Hey 

About the code provided to calculate the UV Index for Grove - UV Sensor GUVA-S12D.

In datasheet we have `UV Index = ((Vout * R2) / ((R1 + R2) * R3 * (10^-9)) - 83) / 21;`. But the coded provided seems to be using the value from `analogRead(A0)` directly which is something between 0 and 1023, and not Vout.

Shouldn't this value be converted to V. Like `Vout = meanVal * (5.0 / 1023.0);`?

```c++
void setup() {
  Serial.begin(9600);
}

void loop() {
  int sensorValue;
  long sum = 0;
  for (int i = 0; i < 1024; i++)  // accumulate readings for 1024 times
  {
    sensorValue = analogRead(A0);
    sum = sensorValue + sum;
    delay(2);
  }
  long meanVal = sum / 1024;  // get mean value
  float sensorVout = meanVal * (5.0 / 1023.0);
  Serial.print("The current UV index is:");
  Serial.println((sensorVout * 1000 / 4.3 - 83) / 21);  // get a detailed calculating expression for UV index in schematic files.
  delay(20);
}
```
